_transcripts/050-rust-1.56-1.57.md

```diff
@@ -0,0 +1,1302 @@
+going to allocate, you know, abort the process, you know, panic the kernel, just
+because they failed to allocate some memory. So this is kind of, maybe step one
+of that process. I'm sure there's much more to go, but so essentially, `Vec`,
+`String`, `HashMap`, `HashSet`, and `VecDeque` all have `try_reserve` as a new
+function.
+
+__Jon__: Yeah, and the big— one of the big missing parts here, I think, for the
+kernel, is that they want to sort of statically guarantee, so guarantee at
+compile time, that there's nothing in there that could panic because of out-of-
+memory, which— this is a step in the right direction, but it doesn't solve the
+problem, right? There is nothing that prevents someone from writing `.push` in
+the kernel, without first calling `try_reserve`. So there's still a question of,
+how can we statically guarantee the things that the Linux kernel requires us to
+guarantee. And that's still being figured out. But at least now we have the
+startings of the mechanisms that the kernel can use. And other use cases too,
+like embedded use cases, can use when they do want to know whether an allocation
+succeeded or failed.
+
+__Jon__: We have a bunch of other stabilized APIs too, which are sort of less,
+monumentus than `try_reserve`. I think the first one I wanted to call out is,
+`Command` now— so this is `std::process::Command`, the thing you use if you want
+to spawn a process inside of a Rust program. So previously, it had, like— you do
+`Command::new`, you give a program name, and then you can add arguments, you can
+set the current directory, you can set environment variables. But you didn't
+have a way to inspect a given `Command` before you execute it, and look at what
+arguments it is being passed, or what environment variables it is being passed.
+Whereas now, it's gained a bunch of `get_` functions, so there's `get_program`,
+`get_args`, `get_envs`, and `get_current_dir`, which gives you the configuration
+of a `Command` and lets you introspect it before you choose to actually execute
+the command.
+
+The other one that I think is pretty neat is that there is now a `map_while`
+method on `Iterator`. So the idea here is that, it's sort of like `filter_map`
+on `Option`. Or I guess not, there's a `filter_map` on `Iterator` too, where it
+will only— it will run each item through a closure, and the `Iterator` will only
+yield the items for which the closure returns `Some` and not the ones where it
+returns `None`. With `map_while`, it's sort of similar, except that it will stop
+the first time it gets `None`. So it's like "take until", and then with a
+closure that when it returns `None`, it stops taking. And the closure also gets
+to map the item, so it's like another nice utility on `Iterator` that makes it
+somewhat easier to write concise `Iterator` sequences, especially if you're sort
+of trying to write your program in a somewhat functional style.
```

Comment:
I think "take until" is in error, and `take_while` is what was intended. I could add an editor's note with a correction? (audio 74:11)

Reply:
Yeah, an editor's note would be good!

Reply:
Done.